Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -51,7 +51,8 @@ public protocol ManifestLoaderProtocol {
     ///   - identityResolver: A helper to resolve identities based on configuration
     ///   - fileSystem: File system to load from.
     ///   - observabilityScope: Observability scope to emit diagnostics.
-    ///   - on: The dispatch queue to perform asynchronous operations on.
+    ///   - delegateQueue: The dispatch queue to call delegate handlers on.
+    ///   - callbackQueue: The dispatch queue to perform completion handler on.
```

Comment:
Is it important for performance to use two queues here, or should this be one queue to ensure that all the delegate methods have completed before the completion closure is called?  For the plugins, that does matter, e.g. receiving all the output before the completion handler is called, but perhaps the situation is different here?  No strong opinion but wanted to check my understanding and assumptions.

I suppose a caller that cares about that can always pass the same queue...

Reply:
If they need to be different, should the second one maybe be called "completionQueue" so it's obvious that it applies to the completion closure parameter?

Reply:
they dont need to be different and in practice they are the same, but the API contract is now clearer and allows the call site to separate between the completion handler queue and the delegate queue _if_ they need to. the API and impl previously intermixed them so the API contract was broken in a way

Reply:
That makes sense to me.